### DISCUSSION TITLE
Basic essential steps and commands to load and show the logo (logo[ ] array)

### DISCUSSION THREAD

**Discussion by th-github** [RudolphRiedel/FT800-FT813#82] [General]
Hi all,

I like to see a list of steps and commands to load and show a simple graphics component. There are two sample arrays in the tft_data.c 
const uint8_t logo[239] PROGMEM 

Or const /*__flash*/ uint8_t pic[3844] PROGMEM

I take it that each byte in this array must be loaded into the BT815 RAM_G. The region looks fit well within the address range of 1Mbyte RAM_G found in datasheet:
#define MEM_PIC1 0x000fa000 /* start of 100x100 pixel test image, ARGB565, needs 20000 bytes of memory */
#define MEM_LOGO 0x000f8000 /* start-address of logo, needs 6272 bytes of memory */
 
Then what follow after loading into the RAM?

I am using STM32F439 and have the Matrix Orbital EVE3 working to show text, circle, rectangle, color. And I need to know how to do the graphics with it. 

Can anyone please post answer here?

Thanks,
Tom

**Comment by RudolphRiedel** [comment]
Hello,

there are two blocks of code in the example to display the graphics, first is in initstaticbackground():
`
/* display the logo */  

EVE_color_rgb(WHITE);  

EVE_cmd_dl(DL_BEGIN | EVE_BITMAPS);  

EVE_cmd_setbitmap(MEM_LOGO, EVE_ARGB1555, 56U, 56U);  

EVE_cmd_dl(VERTEX2F(EVE_HSIZE - 58, 5));  

EVE_cmd_dl(DL_END);  
`

The second is in TFT_display():
`
EVE_color_rgb_burst(WHITE);  

...  

EVE_cmd_setbitmap_burst(MEM_PIC1, EVE_RGB565, 100U, 100U);  

...  

EVE_cmd_dl_burst(DL_BEGIN | EVE_BITMAPS);  

EVE_cmd_dl_burst(VERTEX2F(EVE_HSIZE - 100, (LAYOUT_Y1)));  

EVE_cmd_dl_burst(DL_END);  
`

So what is happenening there in more detail is that the currect graphics object, which is zero in this context since I do not change it,
gets set to the image with CMD_SETBITMAP.
The DL_BEGIN | EVE_BITMAPS tells the co-processor that we want to use graphics primitives and more specifically so bitmaps.
And the VERTEX2F places the bitmap on the screen.

At this point you could place the same bitmap several times on the screen with different coordinates.

That's really it, at least for the simple task to put a bitmap  on the screen.
